RecoTracker/IterativeTracking/python/JetCoreRegionalStep_cff.py

```diff
@@ -254,7 +405,37 @@
 #                                   jetCoreRegionalStepClassifier1,jetCoreRegionalStepClassifier2,
                                    jetCoreRegionalStep)
 JetCoreRegionalStep = cms.Sequence(JetCoreRegionalStepTask)
-seedingDeepCore.toReplaceWith(JetCoreRegionalStep,JetCoreRegionalStep.copyAndExclude([jetCoreRegionalStepHitDoublets]))
+
+JetCoreRegionalStepBarrelTask = cms.Task(jetsForCoreTrackingBarrel,
+                                         firstStepGoodPrimaryVertices,
+                                         #jetCoreRegionalStepClusters,
+                                         jetCoreRegionalStepSeedLayers,
+                                         jetCoreRegionalStepBarrelTrackingRegions,
+                                         jetCoreRegionalStepBarrelHitDoublets,
+                                         jetCoreRegionalStepBarrelSeeds,
+                                         jetCoreRegionalStepBarrelTrackCandidates,
+                                         jetCoreRegionalStepBarrelTracks,
+                                         #                                   jetCoreRegionalStepClassifier1,jetCoreRegionalStepClassifier2,
+                                         jetCoreRegionalStepBarrel)
+JetCoreRegionalBarrelStep = cms.Sequence(JetCoreRegionalStepBarrelTask)
+JetCoreRegionalStepEndcapTask = cms.Task(jetsForCoreTrackingEndcap,
+                                         firstStepGoodPrimaryVertices,
+                                         #jetCoreRegionalStepClusters,
+                                         jetCoreRegionalStepSeedLayers,
+                                         jetCoreRegionalStepEndcapTrackingRegions,
+                                         jetCoreRegionalStepEndcapHitDoublets,
+                                         jetCoreRegionalStepEndcapSeeds,
+                                         jetCoreRegionalStepEndcapTrackCandidates,
+                                         jetCoreRegionalStepEndcapTracks,
+                                         #                                   jetCoreRegionalStepClassifier1,jetCoreRegionalStepClassifier2,
+                                         jetCoreRegionalStepEndcap)
+JetCoreRegionalEndcapStep = cms.Sequence(JetCoreRegionalStepEndcapTask)
```

Comment:
Do you really want to (potentially) filter events?

Reply:
uhmmmm, I'm sorry but I'm not super sure I get this question

the structure was already like that, I simply double the Tasks in order to handle independently the barrel and the endcap regions

sorry for this, but could you please comment further ?

Reply:
opsi
```
jetsForCoreTrackingBarrel = cms.EDFilter('CandPtrSelector', src = cms.InputTag('ak4CaloJetsForTrk'), cut = cms.string('pt > 100 && abs(eta) < 2.5'))
jetsForCoreTrackingEndcap = cms.EDFilter('CandPtrSelector', src = cms.InputTag('ak4CaloJetsForTrk'), cut = cms.string('pt > 100 && abs(eta) > 1.4 && abs(eta) < 2.5'))

```

Reply:
Very sorry, nevermind, I read the configuration too quickly and somehow thought the `jetsForCoreTrackingBarrel` and `jetsForCoreTrackingEndcap` were placed directly in these Sequences. When they're placed in Tasks (as they are), framework implicitly ignores the event filtering decision.

Are the intermediate Sequences `JetCoreRegionalBarrelStep`, `JetCoreRegionalEndcapStep`, and `allJetCoreRegionalStep` needed outside of this file? If not, the replacement logic could be simplified to something along
```py
seedingDeepCore.toReplaceWith(JetCoreRegionalStepTask, cms.Task(
    JetCoreRegionalStepBarrelTask,
    JetCoreRegionalStepEndcapTask
))
```

Reply:
thanks @makortel, I'm going to implement your suggestion